x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_schema.tsx

```diff
@@ -5,43 +5,274 @@
+          className="engineSchemaInnerTable"
+          columns={columns}
+          items={schemaField.indices}
+        />
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+};
 
 export const EngineSchema: React.FC = () => {
   const { engineName } = useValues(EngineIndicesLogic);
   const { isLoadingEngineSchema, schemaFields } = useValues(EngineViewLogic);
   const { fetchEngineSchema } = useActions(EngineViewLogic);
-  const columns: Array<EuiBasicTableColumn<EnterpriseSearchSchemaField>> = [
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [id: string]: JSX.Element;
+  }>({});
+
+  useEffect(() => {
+    fetchEngineSchema({ engineName });
+  }, [engineName]);
+
+  const toggleDetails = (schemaField: SchemaField) => {
+    const newItemIdToExpandedRowMap = { ...itemIdToExpandedRowMap };
+    if (itemIdToExpandedRowMap[schemaField.name]) {
+      delete newItemIdToExpandedRowMap[schemaField.name];
+    } else {
+      newItemIdToExpandedRowMap[schemaField.name] = (
+        <SchemaFieldDetails schemaField={schemaField} />
+      );
+    }
+
+    setItemIdToExpandedRowMap(newItemIdToExpandedRowMap);
+  };
+
+  const columns: Array<EuiBasicTableColumn<SchemaField>> = [
+    {
+      render: ({ type }: SchemaField) => {
+        if (type !== CONFLICT) return null;
+        return <EuiIcon type="error" color="danger" />;
+      },
+      width: '2%',
+    },
     {
-      field: 'field_name',
-      name: i18n.translate('xpack.enterpriseSearch.content.engine.schema.field_name.columnTitle', {
+      name: i18n.translate('xpack.enterpriseSearch.content.engine.schema.fieldName.columnTitle', {
         defaultMessage: 'Field name',
       }),
+      render: ({ name, type }: SchemaField) => (
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          {name.includes('.') && <EuiIcon type="sortRight" color="subdued" />}
+          <EuiText size="s" color={type === CONFLICT ? 'danger' : 'primary'}>
+            <p>{name}</p>
+          </EuiText>
+        </EuiFlexGroup>
+      ),
+      width: '43%',
     },
     {
-      field: 'field_type',
-      name: i18n.translate('xpack.enterpriseSearch.content.engine.schema.field_type.columnTitle', {
-        defaultMessage: 'Field Type',
+      name: i18n.translate('xpack.enterpriseSearch.content.engine.schema.fieldType.columnTitle', {
+        defaultMessage: 'Field type',
       }),
+      render: ({ type }: SchemaField) => {
+        if (type === CONFLICT) {
+          return (
+            <EuiFlexGroup gutterSize="s" alignItems="center">
+              <EuiBadge color="danger">
+                <FormattedMessage
+                  id="xpack.enterpriseSearch.content.engine.schema.fieldType.conflict"
+                  defaultMessage="Conflict"
+                />
+              </EuiBadge>
+            </EuiFlexGroup>
+          );
+        }
+
+        return (
+          <EuiFlexGroup gutterSize="s" alignItems="center">
+            <FieldIcon type={type} />
```

Comment:
is there a better way to do this? this doesn't pick up some types i'd expect it to like `integer`

Reply:
Not that I know of, if it's missing some types we may need a utility function to do the transforming :/ 

something like
```
case (type) {
  case 'integer':
    return 'number';
  default:
    return type;
}
```
But that seems less than ideal.